File: hacker_rank/election_winner.py
# Complete the electionWinner function below.

import collections

def electionWinner(votes):
    map = collections.Counter(votes)
    return sorted(map.items(), key=lambda x: (x[1],x[0]))[-1][0]


assert electionWinner(['Alex', 'Michael', 'Harry', 'Dave', 'Michael', 'Victor', 'Harry', 'Alex', 'Mary', 'Mary']) == 'Michael'
assert electionWinner(['Victor','Veronica','Ryan','Dave','Maria','Maria','Farah','Farah','Ryan','Veronica']) == 'Veronica'
assert electionWinner(['Alice','Allison','Alice','Allison','Ashley','Ashley']) == 'Ashley'

File: hacker_rank/election_winner.rb
# Complete the electionWinner function below.
def electionWinner(votes)
    map = {}
    votes.each do |n|
        map[n] = map.fetch(n, 0) + 1
    end
    map.sort_by(&:reverse)&.last[0]
end

require 'test/unit'
extend Test::Unit::Assertions

assert_equal(electionWinner(['Alex', 'Michael', 'Harry', 'Dave', 'Michael', 'Victor', 'Harry', 'Alex', 'Mary', 'Mary']), 'Michael')
assert_equal(electionWinner(['Victor','Veronica','Ryan','Dave','Maria','Maria','Farah','Farah','Ryan','Veronica']), 'Veronica')
assert_equal(electionWinner(['Alice','Allison','Alice','Allison','Ashley','Ashley']), 'Ashley')
